src/components/select/demoSelectHeader/index.html

```diff
@@ -0,0 +1,22 @@
+<div ng-controller="SelectHeaderController" class="md-padding" ng-cloak>
+  <div>
+    <h1 class="md-title">Pick a vegetable below</h1>
+    <div layout="row">
+      <md-input-container>
+        <label>Vegetables</label>
+        <md-select ng-model="selectedVegetables" md-on-close="clearSearchTerm()" multiple>
+          <md-select-header class="demo-select-header">
+            <input ng-model="searchTerm"
+                   type="search"
+                   placeholder="Search for a vegetable.."
+                   class="demo-header-searchbox _md-text">
+          </md-select-header>
+          <md-optgroup label="vegetables">
+            <md-option ng-value="vegetable" ng-repeat="vegetable in vegetables |
```

Comment:
If I want to perform the search on the server side rather than in memory, then I just do:
```ng-repeat="item in getMatches(searchTerm)"```
similar to https://material.angularjs.org/latest/demo/autocomplete ?
An example with an xhr, or fake xhr might be nice in the docs.